File: groups/group6/Fibonacci.js
function fibonacci(n) {
  if (n === 0) return 0;
  if (n === 1) return 1;
  return fibonacci(n - 1) + fibonacci(n - 2);
}

function main(inputs) {
  const raw = String(inputs[0] ?? "").trim();

  if (raw === "") {
    throw new Error("Please enter the number of terms.");
  }

  const terms = parseInt(raw, 10);

  if (isNaN(terms) || !Number.isInteger(terms)) {
    throw new Error("Input must be a whole number.");
  }

  if (terms <= 2) {
    throw new Error("Number of terms must be greater than 2.");
  }

  if (terms > 30) {
    throw new Error("Please enter 30 or fewer terms.");
  }

  const seq = [];
  for (let i = 0; i < terms; i++) {
    seq.push(fibonacci(i));
  }

  return (
    `This program will find all the terms of the Fibonacci numbers.\n` +
    `\n` +
    `Number of terms: ${terms}\n` +
    `\n` +
    `The Fibonacci numbers are:\n${seq.join(", ")}`
  );
}

File: groups/group5/Java-Recursion/Fibonacci/Fibonacci.java
package Fibonacci;

import java.math.BigInteger;
import java.util.ArrayList;

public class Fibonacci {
    public static String process(String[] inputs) throws Exception {
        int terms = Integer.parseInt(inputs[0].trim());
        if (terms <= 2)
            throw new Exception("Input must be greater than 2.");
        BigInteger a = BigInteger.ZERO, b = BigInteger.ONE;
        ArrayList<String> res = new ArrayList<>();
        for (int i = 0; i < terms; i++) {
            if (i == 0)
                res.add(a.toString());
            else if (i == 1)
                res.add(b.toString());
            else {
                BigInteger next = a.add(b);
                res.add(next.toString());
                a = b;
                b = next;
            }
        }
        return "This program will find all the terms of the Fibonacci numbers.\nThe Fibonacci numbers are:\n"
                + String.join(", ", res);
    }
}